tests/checkpoint/test_checkpoint_result_format.py

```diff
@@ -3513,21 +3513,30 @@ def test_pandas_result_format_in_checkpoint_one_expectation_complete_output_flue
     expectation_suite_name = "metrics_exp"
     context.add_expectation_suite(expectation_suite_name=expectation_suite_name)
 
-    context.sources.add_pandas(name="pandas_datasource").add_dataframe_asset(
+    data_frame_asset = context.sources.add_pandas(
+        name="pandas_datasource"
+    ).add_dataframe_asset(
         name="IN_MEMORY_DATA_ASSET",
-        dataframe=pandas_animals_dataframe_for_unexpected_rows_and_index,
+        batch_metadata={
+            "batch_slice": -1,
```

Comment:
It sure looks like you added this line. It is confusing.

Reply:
@NathanFarmer I see what you mean.  You are absolutely correct.  I tried to include the `Batch` metadata in the `add_dataframe_asset()` method, because that is where the `batch_metadata` argument is allowed, but the specific option about the `slice` is confusing, because, like you said, the slicing is not known till later.  But we have a problem here -- because the Checkpoint specified an ephemeral DataAsset, and the new API change exposes the problem with it.  So we need to be able to specify `batch_metadata` as part of the `build_batch_request()`.  This can be done in the present PR or in the very next one.  Let us discuss the best way to proceed.  Thanks!